sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/SearchClientBuilder.java

```diff
@@ -192,8 +194,19 @@ public SearchClientBuilder endpoint(String endpoint) {
      * @throws IllegalArgumentException If {@link AzureKeyCredential#getKey()} is null or empty.
      */
     public SearchClientBuilder credential(AzureKeyCredential credential) {
-        Objects.requireNonNull(credential, "'credential' cannot be null.");
-        this.credential = credential;
+        this.azureKeyCredential = Objects.requireNonNull(credential, "'credential' cannot be null.");
```

Comment:
We should allow for setting `null` values here and only validate in build*Client method.